tests/v2/validation/provisioning/hosted/hosted_provisioning_test.go

```diff
@@ -149,6 +153,18 @@ func (h *HostedClusterProvisioningTestSuite) TestProvisioningHostedAKSCluster()
 			clusterToken, err := clusters.CheckServiceAccountTokenSecret(client, clusterName)
 			require.NoError(h.T(), err)
 			assert.NotEmpty(h.T(), clusterToken)
+
+			updatedCluster, err := nodepools.ScalingAKSNodePools(client, clusterResp, clusterName, cloudCredential.ID)
+			require.NoError(h.T(), err)
+			assert.Equal(h.T(), clusterName, updatedCluster.Name)
+
+			upgradeK8S, err := upgrades.UpgradeK8SVersion(client, clusterResp, updatedCluster.Name, cloudCredential.ID)
```

Comment:
I personally don't think upgrading, and importing an aks cluster be part of the main aks test. Shouldn't these all be different tests? What if we are adding more validations, to a provisioned aks cluster. Are we always going to upgrade, then import a cluster? In my opinion these should be different test cases in the test suite. Or a separate test suite. I'm open to hear other's opinions on this.

Reply:
I'm open to changing it to be different test cases. I initially had it just like this for further design review, but am open to changing to how we see best for sure.

Reply:
On another note we might need to do separate test suites for all providers. When I wrote this, the framework was still very young, so I'm open to changes. Just some suggestions

Reply:
I agree with expanding the tests horizontally. In my opinion, we need the same thing that happened for custom clusters and provisioning clusters. Adding these extra steps will make this hard to combine in the pipelines in long term. F.e. for the HA release if we had two separate tests as creating imported aks and provision aks, then we would be able to glue everything up in CI. The same idea goes for the scaling test as well, but I don't have a strong opinion on that. But creating everything in one test will make this a test where it should've been designed in the CI part.